test/configuration.js

```diff
@@ -167,6 +167,33 @@ describe('configuration', function () {
       chai.config.showDiff = !chai.config.showDiff;
       assert.equal(chai.Assertion.showDiff, chai.config.showDiff);
     });
-    
+  });
+
+  describe('useProxy', function() {
+    var readNoExistentProperty = function() {
+      expect(false).to.be.tue; // typo: tue should be true
+    };
+
+    it('should have default value equal to true', function() {
+      expect(chai.config.useProxy).to.be.true;
+    });
+
+    describe('when true', function() {
+      it('should use proxy unless user\'s environment doesn\'t support', function() {
+        if (typeof Proxy !== 'undefined' || typeof Reflect !== 'undefined') {
```

Comment:
@lucasfcosta I'm not very happy with an `if` inside the unit test to. This is making the test, test a thing or another depending on a condition, which is very strange. But for chai's context, as we talked it might be a good solution.

@meeber Wow, nice catch. You are absolutely right, thank you.